### PR TITLE
model/tplink.rb: remove password in clear text for older TPLink models (#1574)

### PR DESCRIPTION
**Pull request by lwillek**
https://github.com/ytti/oxidized/issues/1574

### CODE REVIEW

File: lib/oxidized/model/tplink.rb [hunk 1]
```diff
@@ -18,13 +18,17 @@ class TPLink < Oxidized::Model
...
+    # remove unwanted paging line
+    cfg.gsub! /Press any key to contine.*/, ''
     # normalize linefeeds
     cfg.gsub! /(\r|\r\n|\n\r)/, "\n"
     # remove empty lines
     cfg.each_line.reject { |line| line.match /^[\r\n\s\u0000#]+$/ }.join
   end
 
   cmd :secret do |cfg|
+    cfg.gsub! /enable password (\S+)/, 'enable password <secret hidden>'
```
Comment (wk) [inline lib/oxidized/model/tplink.rb]: Do these statements always appear the the beginning of the line, or are they part of other commands?
Could we prefix them with ^ to ensure we don't catch these words appearing elsewhere?

File: lib/oxidized/model/tplink.rb [hunk 2]
```diff
@@ -18,13 +18,17 @@ class TPLink < Oxidized::Model
   end
 
   cmd :all do |cfg|
+    # remove unwanted paging line
+    cfg.gsub! /Press any key to contine.*/, ''
```
Comment (wk) [inline lib/oxidized/model/tplink.rb]: Normally, we'd expect the pager handler at L8 to discard this after handling the pager. If the pager handler didn't match, then no key would've normally been pressed by the pager handler and the model should hang - whereas if it was, the prompt should've been removed. 

What situation leads to this being needed?

File: lib/oxidized/model/tplink.rb (hunk 1 above)
Comment (lwillek) [reply lib/oxidized/model/tplink.rb]: I cant say this for all models, that's why I initially did not wrote a better regex, i.e using '^'. At least for the tree different models I own, the term 'enable password (\S+)' is not part of other commands.
However, the regex `/^enable password (\S+)/` should work too and is more specific, so I will try it.

File: lib/oxidized/model/tplink.rb (hunk 2 above)
Comment (lwillek) [reply lib/oxidized/model/tplink.rb]: I can't answer exactly. I experienced this behave for all the different tplink switches I own, and this line (sometimes followed by one or more special symbols) was part of the output, and the output did not hung at his point. (i.e. the config was always complete)
As you, I think the pager handler did something wrong, but I don't have the knowledge to fix this.
That was the reason for me just to remove the line in the output. I think this should be safe.

Comment (lwillek) [reply lib/oxidized/model/tplink.rb]: However, I just realized that the current regex does not match all situations. So I will review and adapt.

File: lib/oxidized/model/tplink.rb (hunk 1 above)
Comment (lwillek) [reply lib/oxidized/model/tplink.rb]: After review: yes, its possible to use this regex. Other firmware versions use a different approach (`enable secret ....` in Firmware from 2014, encrypted output) (`enale secret` in firmware starting from 2015, encrypted output), so its fine to use the better regex.

**Pull request merged**